packages/mdc-switch/component.ts

```diff
@@ -96,23 +101,30 @@ export class MDCSwitch extends MDCComponent<MDCSwitchFoundation> implements MDCR
 
   private createRipple_(): MDCRipple {
     const {RIPPLE_SURFACE_SELECTOR} = MDCSwitchFoundation.strings;
-    const rippleSurface = this.root_.querySelector(RIPPLE_SURFACE_SELECTOR) as HTMLElement;
+    const rippleSurface =
+        this.root_.querySelector(RIPPLE_SURFACE_SELECTOR) as HTMLElement;
 
-    // DO NOT INLINE this variable. For backward compatibility, foundations take a Partial<MDCFooAdapter>.
-    // To ensure we don't accidentally omit any methods, we need a separate, strongly typed adapter variable.
+    // DO NOT INLINE this variable. For backward compatibility, foundations take
+    // a Partial<MDCFooAdapter>. To ensure we don't accidentally omit any
+    // methods, we need a separate, strongly typed adapter variable.
     const adapter: MDCRippleAdapter = {
       ...MDCRipple.createAdapter(this),
       addClass: (className: string) => rippleSurface.classList.add(className),
       computeBoundingRect: () => rippleSurface.getBoundingClientRect(),
-      deregisterInteractionHandler: (evtType: EventType, handler: EventListener) => {
-        this.nativeControl_.removeEventListener(evtType, handler, applyPassive());
-      },
+      deregisterInteractionHandler:
+          ((evtType: EventType, handler: EventListener) => {
+            this.nativeControl_.removeEventListener(
+                evtType, handler, applyPassive());
+          }) as any,
```

Comment:
What's the purpose of `as any` for the interaction handlers?

Reply:
/cc @mprobst

Reply:
This code fails to compile with TypeScript's `nostrictGenericChecks: false` flag (i.e. what you get from `tsc --strict` by default).

I would have left the error message as a TODO, but it's a bit bulky:

```
material_components_web/switch/component.ts:107:7 - error TS2322: Type '(evtType: "input" | "progress" | "select" | "change" | "abort" | "animationcancel" | "animationend" | "animationiteration" | "animationstart" | "auxclick" | "blur" | "cancel" | "canplay" | "canplaythrough" | "click" | "close" | "contextmenu" | "cuechange" | "dblclick" | "drag" | "dragend" | "dragenter" | "dragexit" | "dragleave" | "dragover" | "dragstart" | "drop" | "durationchange" | "emptied" | "ended" | "error" | "focus" | "focusin" | "focusout" | "gotpointercapture" | "invalid" | "keydown" | "keypress" | "keyup" | "load" | "loadeddata" | "loadedmetadata" | "loadend" | "loadstart" | "lostpointercapture" | "mousedown" | "mouseenter" | "mouseleave" | "mousemove" | "mouseout" | "mouseover" | "mouseup" | "pause" | "play" | "playing" | "pointercancel" | "pointerdown" | "pointerenter" | "pointerleave" | "pointermove" | "pointerout" | "pointerover" | "pointerup" | "ratechange" | "reset" | "resize" | "scroll" | "securitypolicyviolation" | "seeked" | "seeking" | "selectionchange" | "selectstart" | "stalled" | "submit" | "suspend" | "timeupdate" | "toggle" | "touchcancel" | "touchend" | "touchmove" | "touchstart" | "transitioncancel" | "transitionend" | "transitionrun" | "transitionstart" | "volumechange" | "waiting" | "wheel", handler: EventListener) => void' is not assignable to type '<K extends "input" | "progress" | "select" | "change" | "abort" | "animationcancel" | "animationend" | "animationiteration" | "animationstart" | "auxclick" | "blur" | "cancel" | "canplay" | "canplaythrough" | "click" | "close" | "contextmenu" | "cuechange" | "dblclick" | "drag" | "dragend" | "dragenter" | "dragexit" | "dragleave" | "dragover" | "dragstart" | "drop" | "durationchange" | "emptied" | "ended" | "error" | "focus" | "focusin" | "focusout" | "gotpointercapture" | "invalid" | "keydown" | "keypress" | "keyup" | "load" | "loadeddata" | "loadedmetadata" | "loadend" | "loadstart" | "lostpointercapture" | "mousedown" | "mouseenter" | "mouseleave" | "mousemove" | "mouseout" | "mouseover" | "mouseup" | "pause" | "play" | "playing" | "pointercancel" | "pointerdown" | "pointerenter" | "pointerleave" | "pointermove" | "pointerout" | "pointerover" | "pointerup" | "ratechange" | "reset" | "resize" | "scroll" | "securitypolicyviolation" | "seeked" | "seeking" | "selectionchange" | "selectstart" | "stalled" | "submit" | "suspend" | "timeupdate" | "toggle" | "touchcancel" | "touchend" | "touchmove" | "touchstart" | "transitioncancel" | "transitionend" | "transitionrun" | "transitionstart" | "volumechange" | "waiting" | "wheel">(evtType: K, handler: SpecificEventListener<K>) => void'.
  Types of parameters 'handler' and 'handler' are incompatible.
    Types of parameters 'evt' and 'evt' are incompatible.
      Type 'Event' is not assignable to type 'GlobalEventHandlersEventMap[K]'.
        Type 'Event' is not assignable to type 'Event & ProgressEvent<EventTarget> & UIEvent & AnimationEvent & MouseEvent & FocusEvent & DragEvent & ErrorEvent & PointerEvent & KeyboardEvent & SecurityPolicyViolationEvent & TouchEvent & TransitionEvent & WheelEvent'.
          Type 'Event' is missing the following properties from type 'ProgressEvent<EventTarget>': lengthComputable, loaded, total

107       deregisterInteractionHandler: (evtType: EventType, handler: EventListener) => {
          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  material_components_web/ripple/adapter.d.ts:41:5
    41     deregisterInteractionHandler<K extends EventType>(evtType: K, handler: SpecificEventListener<K>): void;
```

(and similarly so for `registerInteractionHandler`)

Reply:
Our tsconfig [has that option set to `true`](https://github.com/material-components/material-components-web/blob/master/tsconfig-base.json#L14), not `false`. If we need our code to pass strict generic checks, wouldn't it make more sense to flip that option and then fix the type errors?

The best way to fix this instance would be to correct the type signature:
```ts
      deregisterInteractionHandler: <K extends EventType>(evtType: K, handler: SpecificEventListener<K>) => {
        this.nativeControl_.removeEventListener(evtType, handler, applyPassive());
      },
```

There are more places than mdc-switch that fail strict generic checks.